src/API/src/db/migrations/1673256699510-create_dataset.ts

```diff
@@ -0,0 +1,22 @@
+import { MigrationInterface, QueryRunner } from "typeorm";
+
+export class createDataset1673256699510 implements MigrationInterface {
+    name = 'createDataset1673256699510'
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(`CREATE TABLE "dataset" ("id" character varying(256) NOT NULL, "status" character varying, "lastUpdatedBy" character varying, "lastUpdatedTime" TIMESTAMP WITH TIME ZONE, "lastReviewedBy" character varying, "lastReviewedTime" TIMESTAMP WITH TIME ZONE, CONSTRAINT "PK_36c1c67adb3d1dd69ae57f18913" PRIMARY KEY ("id"))`);
+        await queryRunner.query(`ALTER TABLE "occurrence" ADD "datasetId" character varying(256) NOT NULL`);
+        await queryRunner.query(`ALTER TABLE "bionomics" ADD "datasetId" character varying(256) NOT NULL`);
+        await queryRunner.query(`ALTER TABLE "occurrence" ADD CONSTRAINT "FK_1350aeebd51e784661ee8adbec2" FOREIGN KEY ("datasetId") REFERENCES "dataset"("id") ON DELETE NO ACTION ON UPDATE NO ACTION`);
+        await queryRunner.query(`ALTER TABLE "bionomics" ADD CONSTRAINT "FK_7110b55018b4e1a919bada31ab1" FOREIGN KEY ("datasetId") REFERENCES "dataset"("id") ON DELETE NO ACTION ON UPDATE NO ACTION`);
```

Comment:
This migration looks good, however when it's run on existing data I think it will fail. It's inserting a new non-nullable column into the bionomics and occurrence tables, so all existing rows will need a value in this column. What we'll need to do, after line 7, is add a new line adding a row into the new dataset table for all existing data to be linked to. Set the id as something like 'initial-data', and the other fields can be left blank. You'll then need to add the datasetId column to bionomics and occurrence as nullable (so remove the `NOT NULL`, set the value of this new column in all the existing rows as the one you've just created, and then change the column to be NOT NULL. See the reference-numeric-id migration for a similar thing.